apps/src/dance/DanceAiClient.ts

```diff
@@ -0,0 +1,101 @@
+import UntypedCachedPalettes from '@cdo/static/dance/ai/model/cached-spacy-palette-map.json';
+import UntypedCachedBackgrounds from '@cdo/static/dance/ai/model/cached-spacy-background-map.json';
+import UntypedCachedForegrounds from '@cdo/static/dance/ai/model/cached-spacy-foreground-map.json';
+
+// Custom types for vector mapping JSONs to play nice with Typescript
+type EmojiAssociation = {
+  association: number[];
+};
+
+type CachedMapping = {
+  emojiAssociations: {[key: string]: EmojiAssociation};
+  output: string[];
+};
+
+const CachedPalettes: CachedMapping = UntypedCachedPalettes;
+const CachedBackgrounds: CachedMapping = UntypedCachedBackgrounds;
+const CachedForegrounds: CachedMapping = UntypedCachedForegrounds;
+
+// Output: JSON containing {backgroundColor:string, backgroundEffect:string, foregroundEffect:string}
+// ToDo: Break down into helper functions for each step rather than repeat code for each field
+export default function cachedAi(input: string) {
+  const inputWords: string[] = input.split(',');
+  const trimmedInput: string[] = inputWords.map(value => value.trim());
+  const inputPaletteVectors: {[key: string]: any} = {};
+  const inputBackgroundVectors: {[key: string]: any} = {};
+  const inputForegroundVectors: {[key: string]: any} = {};
+
+  const {emojiAssociations: paletteAssociations, output: palettes} =
+    CachedPalettes;
+  const {emojiAssociations: backgroundAssociations, output: backgrounds} =
+    CachedBackgrounds;
+  const {emojiAssociations: foregroundAssociations, output: foregrounds} =
+    CachedForegrounds;
+
+  // Pull individual association vectors for each input emoji
+  trimmedInput.forEach(value => {
+    if (Object.prototype.hasOwnProperty.call(paletteAssociations, value)) {
```

Comment:
This seems more complicated than it needs to be. Would a simple loop that iterates over the cached backgrounds, foregrounds, etc work?

Reasoning: This feels a bit esoteric. When I see Object.prototype.hasOwnProperty, it makes me wonder if there's some code smell here. I don't see this sequence used very often, and I definitely had to google it. I get the feeling I'm not the only one since the [mdn docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty) have the line "Here be dragons" as they explain it 😅. 

I could see something like this working:
```
chosenEmojis = [cool, magic, wavy]
foregroundMap = foregroundValues.json
// if the json is reformatted as I recommend in a later comment, the next step isn't needed
reformattedForegroundValues = {}
Object.keys(foregroundMap.emojiAssociations).forEach((emoji) => {
  reformattedForegroundValues[emoji] = {}
  foregroundMap.emojiAssociations[emoji].association.forEach((calculatedValue, index) => {
      reformattedForegroundValues[emoji][foregroundMap.output[index]] = calculatedValue
  })
})
// ^^ then same for reformatting the other files
// end code that could be skipped

// Here, we have a few options. 

// If we want a list of values so we can choose a weighted random value:
var summedForegroundValues = [];
var foregroundEffects = [];
Object.keys(reformattedForegroundValues[chosenEmojis[0]]).forEach((effect) => {
    summedForegroundValues2.push(reformattedForegroundValues[chosenEmojis[0]][effect] + reformattedForegroundValues[chosenEmojis[1]][effect] + reformattedForegroundValues[chosenEmojis[2]][effect])
    foregroundEffects.push(effect)
})

// Or, if we want a mapping of emoji to value
var summedForegroundValues2 = {}
Object.keys(reformattedForegroundValues[chosenEmojis[0]]).forEach((effect) => {
  summedForegroundValues[effect] = reformattedForegroundValues[chosenEmojis[0]][effect] + reformattedForegroundValues[chosenEmojis[1]][effect] + reformattedForegroundValues[chosenEmojis[2]][effect]
})

// Or, if we just want the highest value
var highestValue = Object.keys(reformattedForegroundValues[chosenEmojis[0]]).reduce((highestValue, effect) => {
  return Math.max(highestValue, reformattedForegroundValues[chosenEmojis[0]][effect] + reformattedForegroundValues[chosenEmojis[1]][effect] + reformattedForegroundValues[chosenEmojis[2]][effect])
}, 0)

```

Reply:
It took me a bit to get where you coming from but I think I have a formatting solution for this; I think originally we wanted to be able to separately access the vector of values and any generated text that came with it. However, now that the JSONs are JUST vectors of values, we can sum them all directly.

Reply:
Documenting this follow-up task here https://trello.com/c/Xt0uC2ai